020-Getting-started/070-nextjs.mdx

```diff
@@ -164,43 +191,35 @@ export default function PostsListPage() {
 }
 ```
 
-This error is because Xata stores date fields as an object. To use it within this application we'll need to serialize it to a string before it can be used.
-
-```tsx {6}
-// ...
-records: records.map((record) => ({
-  ...record,
-  // We need to convert our date from an object to a string to use it here
-  // If you didn't set up your pubDate column as "not null" Typescript may complain, so we'll add ? after pubDate for this example
-  pubDate: record.pubDate?.toDateString(),
-})),
-// ...
-```
-
 On save our page should contain no errors. The last thing to do is pass our newly created server-side props to the render so we can see the results on the page.
 
-```tsx {16-32}
-import { getXataClient, Posts } from '../xata';
+```tsx {17-36}
+import { JSONData } from "@xata.io/client";
+import { getXataClient, PostsRecord } from "../xata";
+// We'll assign the client to xata to make it easier to call
 const xata = getXataClient();
 
+// We will make the query to Xata on the server.
+// This will allow the query to read the API_KEY from the .env file
 export const getServerSideProps = async () => {
-  const records: Posts[] = await xata.db.Posts.getMany();
+  const records = await xata.db.Posts.getMany();
   return {
     props: {
-      records: records.map((record) => ({
-        ...record,
-        pubDate: record.pubDate?.toDateString()
-      }))
-    }
+      records: records.map((record) => record.toSerializable()),
+    },
   };
 };
 
 // Pass the props defined above to the page and map against it
-export default function PostsListPage({ records }: { records: Posts[] }) {
+export default function PostsListPage({
+  records,
+}: {
+  records: JSONData<PostsRecord>[];
```

Comment:
I think this will need explanation. Why is `JSONData` necessary?

Reply:
I added a paragraph explanation:

> Note that the type on returned by `toSerializable()` is `JSONData<PostsRecord>`. This essentially means: has all the columns of the Posts table, but the values are JSON serializable. This means they can be passed to the browser while still keeping the type safety of TypeScript.

and asked @SferaDev to review it.